File: JOBCOL FrontEnd/src/services/postService.js
import api from './api';

export const getAllPosts = async () => {

  const response = await api.get('/posts');

  return response.data;
};

export const createPost = async (
  userId,
  description,
  files = []
) => {

  const formData = new FormData();

  formData.append('userId', userId);
  formData.append('description', description);

  files.forEach(file => {
    formData.append('files', file);
  });

  const response = await api.post(
    '/posts',
    formData,
    {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    }
  );

  return response.data;
};

export const likePost = async (postId, userId) => {

  const response = await api.post(
    `/posts/${postId}/like?userId=${userId}`
  );

  return response.data;
};

export const commentPost = async (
  postId,
  userId,
  content
) => {

  const response = await api.post(
    `/posts/${postId}/comment?userId=${userId}&content=${encodeURIComponent(content)}`
  );

  return response.data;
};

File: JOBCOL BackEnd/backend/src/main/java/com/jobcol/backend/PostService/service/PostService.java
package com.jobcol.backend.PostService.service;

import java.util.List;

import org.springframework.web.multipart.MultipartFile;

import com.jobcol.backend.shared.dto.PostDTO;

public interface PostService {
    PostDTO createPost(Long userId,String description,List<MultipartFile> files);

    List<PostDTO> getAllPosts();

    PostDTO getPostById(Long postId);

    void likePost(Long postId, Long userId);

    void commentPost(Long postId, Long userId, String content);
}
